packages/additionalapi/src/controllers/AdditionalApi.ts

```diff
@@ -1,31 +1,40 @@
 import { Request, Response } from "express"
 import { AdditionalApiContext } from "../main.js"
-import { HttpSuccessCodes, lionwebResponse, logger } from "@lionweb/repository-common"
+import {
+    EMPTY_SUCCES_RESPONSE,
+    getIntegerParam,
+    HttpClientErrors,
+    HttpSuccessCodes,
+    isResponseMessage,
+    lionwebResponse,
+    logger
+} from "@lionweb/repository-common"
 
 export interface AdditionalApi {
-    getNodeTree(req: Request, response: Response): void
+    getNodeTree(request: Request, response: Response): void
 }
 
 export class AdditionalApiImpl implements AdditionalApi {
     constructor(private context: AdditionalApiContext) {
     }
     /**
      * Get the tree with root `id`, for one single node
-     * @param req
+     * @param request
      * @param response
      */
-    getNodeTree = async (req: Request, response: Response): Promise<void> => {
-        const idList = req.body.ids
-        let depthLimit = Number.parseInt(req.query["depthLimit"] as string)
-        if (isNaN(depthLimit)) {
-            depthLimit = 99
+    getNodeTree = async (request: Request, response: Response): Promise<void> => {
+        const idList = request.body.ids
+        const depthLimit = getIntegerParam(request, "depthLimit", Number.MAX_SAFE_INTEGER)
+        if (isResponseMessage(depthLimit)) {
```

Comment:
here it is not obvious what `isRespondeMessage` does

Reply:
_isResponseMessage_ is a typeguard that checks whether the parameter object is of type _ResponseMessage_. 
The _getIntegerParam_ method on the line before either returns the value of the requested parameter, or **if the parameter isn't there** a _ResponseMessage_.
So the if checks whether the parameter has been correctly found. 

I agree that the naming is a bit confusing.